src/ctapipe/image/modifications.py

```diff
@@ -100,6 +111,195 @@ def _smear_psf_randomly(
+    must be produced with the same telescope array configuration (and other
+    simulation settings) as the file to which the noise is to be added,
+    but containing only NSB noise (electronic noise should be switched off)
+
+    The number of simulated noise events per telescope in the NSB file must
+    be at least twice the number of waveforms ("nsb_level") from that file
+    that we want to add up. If the NSB file is produced with a level of 25%
+    of dark NSB, and we want to simulate 10x dark NSB, then nsb_level=40 (
+    =10/0.25) and the file must contain at least 80 events. This is to
+    guarantee that the different noise waveforms are not too correlated
+
+    This class can eventually use as NSB file a real data DL0 file which
+    contains pedestal events.
+
+    """
+
+    nsb_file = Path(
+        default_value=None,
+        help="Path to a dedicated NSB-only file (e.g. from sim_telarray)",
+    ).tag(config=True)
+
+    nsb_level = Int(
+        default_value=1,
+        help=(
+            "Number of random instances of the NSB waveforms from the "
+            "NSB file to be added up to the waveform"
+        ),
+    ).tag(config=True)
+
+    n_noise_realizations = Int(
+        default_value=100,
+        help=(
+            "Number of different realizations of the total NSB waveform to "
+            "be created per pixel"
+        ),
+    ).tag(config=True)
+
+    rng_seed = Int(default_value=1, help="Seed for the random number generator").tag(
+        config=True
+    )
+
+    nsb_event_types = List(
+        default_value=[EventType.SKY_PEDESTAL],
+        help="List of event types from which to get the noise waveforms",
+    ).tag(config=True)
+
+    total_noise = dict()
+    # One key per tel_id, each of them is an array of shape
+    # [n_noise_realizations, ngains, npixels, nsamples]
+
+    def __init__(
+        self,
+        subarray: SubarrayDescription,
+        config=None,
+        parent=None,
+        **kwargs,
+    ):
+        """
+
+        Parameters
+        ----------
+        subarray: SubarrayDescription
+            Description of the subarray. Provides information about the
+            camera which are useful in calibration. Also required for
+            configuring the TelescopeParameter traitlets.
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+
+        """
+
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+        self.rng = np.random.default_rng(self.rng_seed)
+
+        event_type_filter = EventTypeFilter(
+            allowed_types=self.nsb_event_types, parent=self
```

Comment:
This shadows the `WaveformModifier.EventTypeFilter.allowed_types` setting in the config. Probably best to just use that and not its own setting? Otherwise, might cause confusion.  Generally there is no reason to pass in settings explicitly inside a Component.

Reply:
> This shadows the `WaveformModifier.EventTypeFilter.allowed_types` setting in the config. Probably best to just use that and not its own setting? Otherwise, might cause confusion. Generally there is no reason to pass in settings explicitly inside a Component.

The problem is I want to have a default value for allowed_types, specific to this class. See comment above by Max: https://github.com/cta-observatory/ctapipe/pull/2847#discussion_r2394359469, apparently that is not simple

Reply:
Alternatively, I could go back to not using EventTypeFilter, and keep the list of allowed types specific to this class.

Reply:
Ah, yes, that's an issue since we can't change defaults.  What I did for other classes like QualityQuery is to create a blank subclass with the new defaults, e.g.:

```
class NoiseEventTypeFilter(EventTypeFilter):
    allowed_types =  allowed_types = Set(
        trait=UseEnum(EventType),
        default_value=[EventType.SKY_PEDESTAL],
        allow_none=True,
        help=(
            "The allowed types. Set to None to allow all types."
            f"Possible values: {_values}."
        ),
    ).tag(config=True)
```

Then use that class.  I think the config sustem should still then accept either `WaveformModifier.EventTypeFilter.allowed_types` or `WaveformModifier.NoiseEventTypeFilter.allowed_types` in your config file.

Reply:
I think the simpler solution here would be using a default function to override the default instead of redefining the tratilet:

```
class NoiseEventTypeFilter(EventTypeFilter):
    @default
     def allowed_types_default(self):
        return {EventType.SKY_PEDESTAL, EventType.PEDESTAL}
```

Reply:
I adopted this solution, I think now it can be set from config, i.e. I tested that this works:
```
config=Config({"WaveformModifier":{"NoiseEventTypeFilter":{"allowed_types":[EventType.FLATFIELD]}}})
wfm = WaveformModifier(subarray=source.subarray, nsb_file=nsb_only_file, nsb_level=3, config=config)
```